562 Data

Mornix opened the pull request:
This PR is part of #562. It adds:
* A tree (chain at the moment) of domains to support functional requirements. `srsDom` -> `reqDom` -> `funcReqDom`
* Generic convenience functions to convert ConceptInstances to Contents. `mkConCC`, `mkConCC'`, and `mkConCC''`. The latter two are more "convenient" uses of the first.

JacquesCarette commented:
The master failures should be fixed -- please merge, that should allow this one to then be merged in as well.

Mornix commented:
You are absolutely correct, `mkConCC*` were in the wrong place for how specific they are. I moved them to DocumentLanguage. I also renamed `mkConCC''` to `mkEnumCC` since it is a targeted specialization to make `Contents` from the `Enumeration` constructor.
